cannon/mipsevm/tests/evm_common_test.go

```diff
@@ -206,10 +216,10 @@ func TestEVMSingleStep_Add(t *testing.T) {
 				expected.NextPC = 8
 
 				if tt.ifImm {
-					expected.Registers[8] = tt.expectImm
+					expected.Registers[8] = tt.expectRes
 					expected.Registers[17] = tt.rs
 				} else {
-					expected.Registers[8] = tt.expectRD
+					expected.Registers[8] = tt.expectRes
 					expected.Registers[17] = tt.rs
 					expected.Registers[18] = tt.rt
 				}
```

Comment:
The call to `expected := testutil.NewExpectedState(state)` copies over existing register values into `expected`.

So, should be able to simplify this to:
```
expected.Registers[8] = tt.expectRes
```